src/libraries/System.IO.Pipes/src/System/IO/Pipes/NamedPipeClientStream.cs

```diff
@@ -193,6 +195,19 @@ public Task ConnectAsync(int timeout, CancellationToken cancellationToken)
             return Task.Run(() => ConnectInternal(timeout, cancellationToken, startTime), cancellationToken);
         }
 
+        public Task ConnectAsync(TimeSpan timeout, CancellationToken cancellationToken = default) =>
+            ConnectAsync(ToTimeoutMilliseconds(timeout), cancellationToken);
+
+        private static int ToTimeoutMilliseconds(TimeSpan timeout)
+        {
+            long totalMilliseconds = (long)timeout.TotalMilliseconds;
+            if (totalMilliseconds < -1 || totalMilliseconds > int.MaxValue)
+            {
+                throw new ArgumentOutOfRangeException(nameof(timeout));
```

Comment:
If possible and not too verbose, we should have unit tests for both conditions that cause this exception, on every library that now has one of these ToTimeoutXXX methods in it. Hopefully these are generally 1-2 line tests.

Reply:
I see you have several cases, but not for both < -1 and > int.MaxValue in all cases.